api/client/sg.cc

```diff
@@ -435,53 +435,91 @@ static int sub_sg_rule_del(int argc, char **argv,
     }
 
     string sg = string(argv[4]);
-    string hash = string(argv[5]);
-    string req =
-        "messages {"
-        "  revision: " PROTO_REV
-        "  message_0 {"
-        "    request {"
-        "      sg_rule_list: \"" + sg + "\""
-        "    }"
-        "  }"
-        "}";
-
-    proto::Messages res;
-    if (request(req, &res, options, false) || check_request_result(res))
-        return 1;
+    if (sizeof(string(argv[5])) == 16) {
```

Comment:
what are you trying to do here ?
for me, this code check the size of std::string, and it really look like it check the size of an std::string
So this is obviously a very tricky code you are using here(and compiler dependent), maybe not what you wanted to do ? and if you really want to do that, please add a comment to explain what you are doing(or a macro, or a function)